### PR TITLE
Add ProxyGet to node expansion in client-go

### PR DESCRIPTION
**Pull request by siadat**
I need this to be able to access the "/stats/container" endpoint using client-go.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36367)

### CODE REVIEW

**Comment by siadat** [comment]
@k8s-bot ok to test

**Comment by siadat** [comment]
Let me know if I need to change anything else! :)
@dchen1107

**Comment by siadat** [comment]
@timstclair My goal is to implement [your answer](http://stackoverflow.com/a/35212987) using client-go. Am I doing the right thing?

**Comment by siadat** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/36367)*

**Comment by siadat** [comment]
ping

File: staging/src/k8s.io/client-go/kubernetes/typed/core/v1/node_expansion.go [hunk 1]
```diff
@@ -41,3 +45,16 @@ func (c *nodes) PatchStatus(nodeName string, data []byte) (*v1.Node, error) {
 		Into(result)
 	return result, err
 }
+
+// ProxyGet returns a response of the node by calling it through the proxy.
+func (c *nodes) ProxyGet(scheme, name, port, path string, params map[string]string) rest.ResponseWrapper {
+	request := c.client.Get().
+		Prefix("proxy").
+		Resource("nodes").
+		Name(net.JoinSchemeNamePort(scheme, name, port)).
```
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/kubernetes/typed/core/v1/node_expansion.go]: @liggitt Does the node proxy support `scheme`? Shall we default it to "" here?

Comment (liggitt) [inline staging/src/k8s.io/client-go/kubernetes/typed/core/v1/node_expansion.go]: It supports it. https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/node/strategy.go#L177

File: staging/src/k8s.io/client-go/kubernetes/typed/core/v1/node_expansion.go [hunk 2]
```diff
@@ -41,3 +45,16 @@ func (c *nodes) PatchStatus(nodeName string, data []byte) (*v1.Node, error) {
 		Into(result)
 	return result, err
 }
+
+// ProxyGet returns a response of the node by calling it through the proxy.
+func (c *nodes) ProxyGet(scheme, name, port, path string, params map[string]string) rest.ResponseWrapper {
+	request := c.client.Get().
+		Prefix("proxy").
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/kubernetes/typed/core/v1/node_expansion.go]: This should use a subresource, not a prefix

**Pull request closed without merging**